### PR TITLE
Add getModuleIndex to Logic

### PR DESCRIPTION
**Pull request by UncleGrandpa925**
Hi @simonteozw , can you quickly review this PR for me 😄 It should be pretty short! Thanks a lot!

### CODE REVIEW

File: src/main/java/trackitnus/model/Model.java [hunk 1]
```diff
@@ -256,16 +256,22 @@
     ObservableList<Lesson> getModuleLessons(Code code);
 
     /**
-     * @param lesson The lesson to check for.
+     * @param lesson The lesson to get index
```
Comment (simonteozw) [inline src/main/java/trackitnus/model/Model.java]: I think this should be changed to `get index of`

File: src/main/java/trackitnus/model/Model.java [hunk 2]
```diff
@@ -256,16 +256,22 @@
...
      * @return The index of the lesson w.r.t the last shown Lesson list.
      */
     Index getLessonIndex(Lesson lesson) throws CommandException;
 
+    /**
+     * @param module The module to get index
+     * @return The index of the lesson in the list of all modules
+     */
+    Index getModuleIndex(Module module) throws CommandException;
```
Comment (simonteozw) [inline src/main/java/trackitnus/model/Model.java]: Same here. `get index of`

**Review by simonteozw — COMMENTED** [review]
Looks good! Just some comments

File: src/main/java/trackitnus/model/Model.java (hunk 2 above)
Comment (UncleGrandpa925) [reply src/main/java/trackitnus/model/Model.java]: fixed!

**Review by simonteozw — APPROVED** [review]
LGTM

File: src/main/java/trackitnus/model/ModelManager.java [hunk 3]
```diff
@@ -323,6 +323,16 @@ public Index getLessonIndex(Lesson lesson) throws CommandException {
         return Index.fromZeroBased(index);
     }
 
+    @Override
+    public Index getModuleIndex(Module module) throws CommandException {
+        ObservableList<Module> moduleList = getFilteredModuleList();
+        int index = moduleList.indexOf(module);
+        if (index == -1) {
+            throw new CommandException(Messages.MESSAGE_MODULE_DOES_NOT_EXIST);
+        }
```
Comment (simonteozw) [inline src/main/java/trackitnus/model/ModelManager.java:332]: Sorry I just realised, should you update the predicate to show all Modules? Or doesn't matter. In task and lesson we will predicate first

**Pull request merged**